src/operator/tensor/broadcast_reduce_op.h

```diff
@@ -166,44 +168,59 @@ inline TShape ReduceAxesShapeImpl(const TShape& ishape, const TShape& axis,
     }
   }
 
+  if (axis[0] < 0) {
```

Comment:
The axis is actually still correctly sorted, even if there is negative entry in the axis.
Since axis[0] is the smallest value, this is checking whether it exceeds the range (length of in_shape) if it's a negative value.